src/App.js

```diff
@@ -134,54 +138,163 @@ class App extends Component {
   }
 
   walk () {
-    const milesGained = rangeGenerator(12, 25)
-    let newMiles = this.state.progress.miles
-    newMiles += milesGained
-    let newDays = this.state.progress.days
-    newDays += 1
-    const peopleList = this.state.people.map((person) => Object.assign({}, person))
-    const peopleLiving = peopleList.filter((character) => character.status !== DEAD)
-    const newPeopleList = this.peopleLoseHealth(peopleLiving)
-    const foodPortions = peopleLiving.length
-    const foodLost = rangeGenerator(2 * foodPortions, 5 * foodPortions)
-    let newFood = this.state.inventory.food
-    if (this.state.inventory.food > 0) {
-      newFood -= foodLost
+    // walk is the only thing that will set state
+    let forSetState
+    const milesGained = randomGenerator(12, 25)
+    const newProgress = Object.assign({}, this.state.progress)
+    newProgress['miles'] += milesGained
+    newProgress['days'] += 1
+    if (newProgress['miles'] > TRAIL_MILES) {
+      forSetState = this.finishGame(newProgress)
+    } else {
+      forSetState = this.continueGame(newProgress)
+    }
+    this.setState(forSetState)
+  }
+
+  finishGame (progress) {
+    const gameStateObject = Object.assign({}, this.state.game)
+    gameStateObject['gameState'] = WIN
+    const progressObject = Object.assign({}, this.state.progress)
+    return {
+      game: gameStateObject,
+      progress: progressObject
+    }
+  }
+
+  continueGame (progressObj) {
+    let newGameMessage = ''
+    const peopleLiving = this.ignoreStatusDead(this.state.people) // removes all dead characters from play
```

Comment:
`const peopleLiving = this.ignoreStatusDead(this.state.people) // removes all dead characters from play`
When I see this, I think, "this function should be called `removeDeadCharactersFromPlay()`